### ISSUE TITLE
xrdp-0.9.3 clipboard not working properly with tigervnc-1.8.0 backend

### ISSUE THREAD

**Issue by imc0** [neutrinolabs/xrdp#873]
System is Fedora 26 with the Fedora-packaged versions of:
xrdp-0.9.3-1.fc26.x86_64
xorgxrdp-0.2.3-1.fc26.x86_64
tigervnc-1.8.0-1.fc26.x86_64
rdesktop-1.8.3-3.fc24.x86_64
freerdp-2.0.0-31.20170724gitf8c9f43.fc26.x86_64

and the session being viewed is GNOME (3.24).

If I select the Xorg backend at login then the clipboard seems to work fine, although the mouse pointer in rdesktop is broken (appears as a black square with a white cursor mask inside it).

If I select the VNC backend then:
* using rdesktop on the viewing system:
  - if I make a primary selection outside the rdesktop window on the viewing system, this usually appears in the clipboard (not the primary selection) of the xrdp session and I can paste it.
  - If I make a primary selection inside the xrdp session, this doesn't affect the clipboard on either side.
  - If I try to copy something into the clipboard inside the xrdp session, the clipboard on both sides is cleared.  This means that I can't even copy/paste within the xrdp session.
* using xfreerdp on the viewing system:
  - If I make a primary selection outside the rdp window, this doesn't affect the clipboard (or the primary selection) of the xrdp session.
  - If I copy something into the clipboard outside the rdp window, this appears in the clipboard of the xdrp session.
  - If I make a primary selection inside the xrdp session, this doesn't affect the clipboard on either side.
  - If I try to copy something into the clipboard inside the xrdp session, xfreerdp crashes with the following messages:
```
[12:44:25:623] [8313:8323] [ERROR][com.freerdp.channels.cliprdr.client] - Invalid short format list length: 24
[12:44:25:623] [8313:8323] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_process_format_list failed with error 1359!
[12:44:25:623] [8313:8323] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_order_recv failed with error 1359!
[12:44:25:623] [8313:8314] [ERROR][com.freerdp.core] - cliprdr_virtual_channel_client_thread reported an error. Error was 1359
[12:44:25:623] [8313:8314] [INFO][com.freerdp.client.x11] - Network disconnect!
[12:44:25:623] [8313:8314] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor
```
After the crash, the xrdp session still exists and can be reconnected (by either xfreerdp or rdesktop).

**Comment by mailinglists35** [comment]
Hi,
is there any workaround for this please?
If I connect directly to the tigervnc server, clipboard works both ways.
If I connect to the tigervnc via xrdp, I can only copy from vnc to client. but I cannot paste anything from client to xrdp/vnc (neither primary nor secondary)
thanks
using 0.9.8 and tiger 1.8.0 and windows10 1803 mstsc.exe

**Comment by mailinglists35** [comment]
or is there any other virtual vnc server that works okay with xrdp?

**Comment by mailinglists35** [comment]
@lnicola xorgxrdp does not allow two clients to share the same session unless a [commit is reverted](https://github.com/neutrinolabs/xorgxrdp/issues/128). I find that too complicated and it would be much easier if this was an option instead.

**Comment by mailinglists35** [comment]
so I've made the effort to try to rebuild from a .src.rpm distrubited by fedora rawhide (the only src.rpm with 0.9.8 that includes the fix for properly unmounting fuse drives at logout)

but no luck:
````
make[3]: Leaving directory `/home/user/rpmbuild/BUILD/xrdp-0.9.8/sesman/tools'
Making all in chansrv
make[3]: Entering directory `/home/user/rpmbuild/BUILD/xrdp-0.9.8/sesman/chansrv'
  CC       chansrv.o
  CC       chansrv_common.o
  CC       chansrv_fuse.o
chansrv_fuse.c: In function 'xfuse_check_wait_objs':
chansrv_fuse.c:565:9: error: too few arguments to function 'fuse_session_process'
         fuse_session_process(g_se, buf, rval, tmpch);
         ^
In file included from chansrv_fuse.c:95:0:
/usr/include/fuse/fuse_lowlevel.h:1688:6: note: declared here
 void fuse_session_process(struct fuse_session *se, const char *buf, size_t len,
      ^
make[3]: *** [chansrv_fuse.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory `/home/user/rpmbuild/BUILD/xrdp-0.9.8/sesman/chansrv'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/user/rpmbuild/BUILD/xrdp-0.9.8/sesman'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/user/rpmbuild/BUILD/xrdp-0.9.8'
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.k6ZmnY (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.k6ZmnY (%build)
````

**Comment by jcarlstedt** [comment]
> Hi,
> is there any workaround for this please?
> If I connect directly to the tigervnc server, clipboard works both ways.
> If I connect to the tigervnc via xrdp, I can only copy from vnc to client. but I cannot paste anything from client to xrdp/vnc (neither primary nor secondary)
> thanks
> using 0.9.8 and tiger 1.8.0 and windows10 1803 mstsc.exe

See fix for #1824

**Comment by metalefty** [comment]
I believe this has been fixed.
